Add ux documentation - part 1 (#185)
SchwarzIT/onyx: apps/docs/src/tokens/introduction.md

# Introduction

> Understanding the logic, structure and usage of tokens is crucial for creating onyx applications. Here is everything you need to know. Please feel free to contact the departments of UX/UI or Web Frontend Enterprise if you got further questions.

Reviewer: ```suggestion
Understanding the logic, structure and usage of tokens is crucial for creating onyx applications. Here is everything you need to know. Please feel free to visit our [Q&A](https://github.com/SchwarzIT/onyx/discussions/categories/q-a) if you have any questions.
```

This does not work for external users, we should only promote our public support channels

## What is a design token?

> A design token is a fundamental, reusable building block for design systems like onyx. It represents a specific aspect of a design , such as colors, typography, spacing or other visual elements, and is defined as a named value or parameter. Design tokens help maintain consistency and streamline the design and development process across various platforms and applications.

Reviewer: ```suggestion
A design token is a fundamental, reusable building block for design systems like onyx. It represents a specific aspect of a design as a key-value pair such as color, typography, spacing or another visual element. Design tokens help maintain consistency and streamlining the design and development process across various platforms and applications.
```

::: tip Important
Design tokens are variables, that are used equally in development and design to ensure the customizability under the consideration of the consistency across all onyx applications.
:::

## How do they work?

> First of all it is essential to understand the basic systematic of tokens. Tokens are leveled into three layers, that are related to each other. Every stage of tokens got its own purpose, but not all stages are used in the design and development process.

Reviewer: ```suggestion
First of all, it is essential to understand the basic systematic of tokens. Tokens are categorised into three layers, that are related to each other. Every token layer has its own responsibility, but not all layers are used directly in the design and development process.
```

### Global tokens

Global tokens are the fundamental elements of every design system to establish a unique design language. This level of tokens define fundamental properties such as colors, spacings and other without any usecase or context. All of those values are stored in a systematic way.

For example `onyx-color-universal-white` is just the value for white color in general.

::: warning Important
Global tokens are not used within the process of designing and developing an application! As a result they are hidden to minimize sources of errors. Imagine them like the invisible base for your product
:::

![Logo](/assets/global_token.png)

### Semantic tokens

> This stage of tokens puts the base token collection into context. At this point, semantic tokens differentiate between light mode and dark mode and connects a usecase or purpose to the values, that were defined in the global token collection. The naming of semantic tokens is less generic and more specific to the intended use.

Reviewer: ```suggestion
This token layer puts the global tokens into context. At this point, semantic tokens differentiate between light and dark mode and connect a usecase or purpose to the values, that were defined in the global tokens. The naming of semantic tokens is less generic and more specific to the intended use.
```

For example `onyx-color-surface-bg-blank` is exclusively used for a blank background color. To have a more detailed view about the color token system and how light and dark modes are implemented, please check out the [color documentation](/basics/colors).

::: warning Important

> Semantic tokens are used from both developers and designers to build applications with existing standard components. With the use of those tokens a seamless switch between light mode and dark mode is guaranteed. Otherwise ist will not work without additional effort of building.

Reviewer: ```suggestion
Semantic tokens are used from both developers and designers to build applications with existing standard components. With the use of those tokens, a seamless switch between light and dark mode is guaranteed. Otherwise, this will not work without additional effort.
```
